src/aal/aal_sparc.h

```diff
@@ -0,0 +1,62 @@
+#pragma once
+
+#if defined(__arch64__) // More reliable than __sparc64__
+#  define SNMALLOC_VA_BITS_64
+#else
+#  define SNMALLOC_VA_BITS_32
+#endif
+
+namespace snmalloc
+{
+  /**
+   * Sparc architecture abstraction layer.
+   */
+  class AAL_Sparc
+  {
+  public:
+    /**
+     * Bitmap of AalFeature flags
+     */
+    static constexpr uint64_t aal_features = IntegerPointers;
+
+    static constexpr enum AalName aal_name = Sparc;
+
+#ifdef SNMALLOC_VA_BITS_64
+    /**
+     * Even Ultra-Sparc I supports 8192 and onwards
+     */
+    static constexpr size_t smallest_page_size = 0x2000;
+#else
+    static constexpr size_t smallest_page_size = 0x1000;
+#endif
+
+    /**
+     * On Sparc ideally pause instructions ought to be
+     * optimised per Sparc processor but here a version
+     * as least common denominator to avoid numerous ifdef,
+     * reading Conditions Code Register here
+     */
+    static inline void pause()
+    {
+      __asm__ volatile("rd %%ccr, %%g0" ::: "memory");
+    }
+
+    static inline void prefetch(void* ptr)
+    {
+#ifdef SNMALLOC_VA_BITS_64
+      __asm__ volatile("prefetch [%0], 0" ::"r"(ptr));
```

Comment:
Can use:
```C
      __asm__ volatile("prefetch %0, 0" :: "m"(*(const char*)ptr));
```
in order to support reg+reg and reg+imm addressing. Technically should be `*(const char (*)[])` (according to clang-format there is a space there even for `PointerAlignment: Left`, which makes sense I guess) to indicate it reads an unknown number of bytes, but given it's just a hint with no side-effects (nor actual value-dependence) it doesn't matter, and Clang doesn't seem to like using `[]` in that context, when GCC does and explicitly documents it as the way to express such constraints.

Probably also worth documenting that the 0 is for fun 0 meaning "Prefetch for several reads".